packages/esm-patient-medications-app/src/active-medications/active-medications.component.tsx

```diff
@@ -12,29 +12,38 @@ interface ActiveMedicationsProps {
 const ActiveMedications: React.FC<ActiveMedicationsProps> = ({ patient }) => {
   const { t } = useTranslation();
   const displayText = t('activeMedicationsDisplayText', 'Active medications');
-  const headerTitle = t('activeMedicationsHeaderTitle', 'active medications');
+  const headerTitle = t('activeMedicationsHeaderTitle', 'Active Medications');
 
-  const { data: activePatientOrders, error, isLoading, isValidating } = usePatientOrders(patient?.id, 'ACTIVE');
+  const { data: activePatientOrders, error, isLoading, isValidating } = usePatientOrders(patient?.id);
 
   const launchAddDrugWorkspace = useLaunchWorkspaceRequiringVisit('add-drug-order');
 
   if (isLoading) return <DataTableSkeleton role="progressbar" />;
 
   if (error) return <ErrorState error={error} headerTitle={headerTitle} />;
 
-  if (activePatientOrders?.length) {
+  const activeMedications = activePatientOrders?.filter((order) => {
+    const isDateStoppedUnset = !order.dateStopped;
+    const isActionValid = order.action !== 'DISCONTINUE';
+    const isAutoExpireDateValid = !order.autoExpireDate || new Date(order.autoExpireDate) > new Date();
+
+    return isDateStoppedUnset && isActionValid && isAutoExpireDateValid;
+  });
```

Comment:
We don't need to do any filtering here. That's already taken care of by the `useActivePatientOrders` hook.